tools/osbuilder/image-builder/image_builder.sh

```diff
@@ -372,6 +372,55 @@ create_disk() {
 	OK "Partitions created"
 }
 
+create_erofs_rootfs_image() {
+	local rootfs="$1"
+	local image="$2"
+	local fs_type="$3"
+	local block_size="$4"
+
+	if [ $block_size -ne 4096 ]; then
+		die "Invalid block size for erofs"
+	fi
+
+	readonly mount_dir=$(mktemp -p ${TMPDIR:-/tmp} -d osbuilder-mount-dir.XXXX)
+
+	info "Copying content from rootfs to root partition"
+	cp -a "${rootfs}"/* "${mount_dir}"
+
+	info "Removing unneeded systemd services and sockets"
```

Comment:
Just curious to what extent this plays a role in the observed reduction in size?

Reply:
mkfs.erofs accepts a src rootdir directory as an input rather than some device, so no need to guess the device dest size first.
Last weekend I observed it by using generated centos 8 rootfs image with the original script and the modified script with new EROFS fs type.
If my kata-container environment and all of my operations done are correct, I think the data is correct too.